### PR TITLE
ISSUE-701 Restart consumption in case of consumer crash

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+        @ProvidesIntoSet
+        InitializationOperation start(ScheduledReconnectionHandler scheduledReconnectionHandler) {
+            return InitilizationOperationBuilder
+                .forClass(ScheduledReconnectionHandler.class)
+                .init(scheduledReconnectionHandler::start);
+        }
+    }
+
+    public interface RabbitMQManagementAPI {
```
Comment (chibenwa) [inline tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java:87]: Needed to relax activityStatus

File: tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+                return queueLength;
+            }
+        }
+
+        class ConsumerDetails {
+            @JsonProperty("consumer_tag")
+            String tag;
+
+            @JsonProperty("activity_status")
+            String status;
```
Comment (chibenwa) [inline tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java:151]: Here

**Review by Arsnael — APPROVED** [review]
Maybe doc missing? otherwise looks good to me

File: tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+import reactor.core.scheduler.Schedulers;
+
+public class ScheduledReconnectionHandler implements Startable {
+    public record ScheduledReconnectionHandlerConfiguration(boolean enabled) {
+        public static ScheduledReconnectionHandlerConfiguration parse(PropertiesProvider propertiesProvider) throws ConfigurationException {
+            try {
+                Configuration configuration = propertiesProvider.getConfiguration("rabbitmq");
+                return new ScheduledReconnectionHandlerConfiguration(configuration.getBoolean("scheduled.consumer.reconnection.enabled", true));
+            } catch (FileNotFoundException e) {
+                return new ScheduledReconnectionHandlerConfiguration(true);
```
Comment (quantranhong1999) [inline tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java:68]: Should be `false` when no file `rabbitmq.properties`?

**Review by Arsnael — APPROVED** [review]

**Pull request merged**